xbmc/addons/addoninfo/AddonType.h

```diff
@@ -107,6 +109,15 @@ class CAddonType : public CAddonExtensions
    */
   static bool IsDependencyType(TYPE type);
 
+  /*!
+   * @brief Indicates whether a given addon should be visible in the gui depending
+   * on its type
+   *
+   * @param[in] addon the given addon
+   * @return true if this addon should be visible in a dialog
+   */
+  static bool IsVisibleInDialog(const std::shared_ptr<IAddon>& addon);
```

Comment:
You're mixing in GUI stuff in a core class. This smells fishy. Maybe it's just bad naming?